### PR TITLE
[YUNIKORN-2722] Expose the IsOriginator flag in REST

### PR DESCRIPTION
**Pull request by blueBlue0102**
### What is this PR for?

Add the `Originator` field to the `Allocation` and `AllocationDAOInfo` structs

### What type of PR is it?
* [x] - Improvement

### What is the Jira issue?
<https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-2722>

### Screenshots (if appropriate)
image

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/allocation_test.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func TestNewAllocFromSI(t *testing.T) {
 	assert.Assert(t, alloc.IsPlaceholder(), "ask should have been a placeholder")
 	assert.Equal(t, alloc.GetTaskGroup(), "testgroup", "TaskGroupName not set as expected")
 	assert.Equal(t, alloc.GetAsk().GetCreateTime(), time.Unix(past, 0)) //nolint:staticcheck
-	assert.Assert(t, alloc.GetAsk().IsOriginator(), "ask should have been an originator")
+	assert.Assert(t, alloc.IsOriginator(), "ask should have been an originator")
```
Comment (chenyulin0719) [inline pkg/scheduler/objects/allocation_test.go:241]: "ask should not have been an originator"
-> 
"allocation should have been an orignator".

Maybe we can check both allocation & ask's IsOriginator()

Can do the same check for placeholder.

File: pkg/scheduler/objects/allocation_test.go [hunk 2]
```diff
@@ -251,7 +251,7 @@ func TestNewAllocFromSI(t *testing.T) {
 	endTime := time.Now().Unix()
 	assert.Assert(t, alloc.GetAsk().GetCreateTime().Unix() >= startTime, "alloc create time is too early")
 	assert.Assert(t, alloc.GetAsk().GetCreateTime().Unix() <= endTime, "alloc create time is too late")
-	assert.Assert(t, !alloc.GetAsk().IsOriginator(), "ask should not have been an originator")
+	assert.Assert(t, !alloc.IsOriginator(), "ask should not have been an originator")
```
Comment (chenyulin0719) [inline pkg/scheduler/objects/allocation_test.go:254]: ditto

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
Overall LGTM, one minor change.

**Comment by blueBlue0102** [comment]
Hi @chenyulin0719 , thank you for your review.
I have made some adjustments. Please take a look.😄

**Review by chenyulin0719 — APPROVED** [review]
LGTM

**Review by pbacsko — APPROVED** [review]
+1

**Pull request closed without merging**